tests/test_agents.py

```diff
@@ -740,15 +778,21 @@ async def test_deepcopy_env(self, agent_test_settings: Settings) -> None:
         env_copy = deepcopy(env)
         assert env.state == env_copy.state
 
-        # 3. Generate an answer for both, and confirm they are identical
+        # 3. Generate an answer and complete for both, and confirm they are identical
         gen_answer_action = ToolRequestMessage(
             tool_calls=[ToolCall.from_name("gen_answer")]
         )
-        _, _, done, _ = await env.step(gen_answer_action)
+        await env.step(gen_answer_action)
+        _, _, done, _ = await env.step(
+            ToolRequestMessage(tool_calls=[ToolCall.from_name("complete")])
+        )
         assert done
         assert not env.state.session.could_not_answer
         assert env.state.session.used_contexts
-        _, _, done, _ = await env_copy.step(gen_answer_action)
+        await env_copy.step(gen_answer_action)
+        _, _, done, _ = await env.step(
```

Comment:
This should have `env_copy.step`, fixed in https://github.com/Future-House/paper-qa/pull/698